naptime/src/test/scala/org/coursera/naptime/TypesTest.scala

```diff
@@ -91,4 +96,44 @@ class TypesTest extends AssertionsForJUnit {
     assert(resultingType.getField("description") != null)
     assert(resultingType.getField("description").getRecord == Course.SCHEMA)
   }
+
+  @Test
+  def relations(): Unit = {
+    val resourceFields = ResourceFields[Empty]
+      .withRelated(
+        "includeOnly" -> ResourceName("includeOnly", 1),
+        "shared" -> ResourceName("shared", 2))
+      .withGraphQLRelations(
+        "gqlOnly" -> FinderGraphQLRelation(ResourceName("gqlOnly", 1), "find"),
+        "shared" -> GetGraphQLRelation(ResourceName("shared", 2), "get")
+      )
+    val resultingType =
+      Types.computeAsymType("Relations", Empty.SCHEMA, Empty.SCHEMA, resourceFields)
+
+    val fieldsExpectedNames = List("id", "includeOnly", "shared", "gqlOnly")
+    assert(resultingType.getFields.asScala.map(_.getName) === fieldsExpectedNames)
+    val includeOnlyField = resultingType.getField("includeOnly")
+    val includeOnlyExpectedProperties = Map(
+      "included" -> ResourceName("includeOnly", 1).toAnnotation.data)
+    assert(includeOnlyField != null)
+    assert(!includeOnlyField.getOptional)
+    assert(includeOnlyField.getType == new ArrayDataSchema(new StringDataSchema))
+    assert(includeOnlyField.getProperties.asScala === includeOnlyExpectedProperties)
+    val sharedField = resultingType.getField("shared")
+    val sharedExpectedProperties = Map(
+      "included" -> ResourceName("shared", 2).toAnnotation.data,
+      "relatedOn" -> GetGraphQLRelation(ResourceName("shared", 2), "get").toAnnotation.data
+    )
+    assert(sharedField != null)
+    assert(sharedField.getOptional)
+    assert(sharedField.getType == new StringDataSchema)
```

Comment:
This seems a bit confusing since it's different from L120 but it's because we're using a Get relation, maybe we can use the same GraphQL relation type in both cases so there's not this inconsistency?

Reply:
I can, but I did it this way because I wanted to exercise both branches of the `match` with `Some`. I can also add a comment explaining the difference.